crates/oxc_linter/src/rules/unicorn/prefer_array_some.rs

```diff
@@ -62,10 +62,24 @@ impl Rule for PreferArraySome {
                     return;
                 }
 
-                ctx.diagnostic(over_method(
-                    // SAFETY: `call_expr_method_callee_info` returns `Some` if `is_method_call` returns `true`.
-                    call_expr_method_callee_info(call_expr).unwrap().0,
-                ));
+                ctx.diagnostic_with_fix(
+                    over_method(
+                        // SAFETY: `call_expr_method_callee_info` returns `Some` if `is_method_call` returns `true`.
+                        call_expr_method_callee_info(call_expr).unwrap().0,
+                    ),
+                    |fixer| {
+                        fixer.replace(
+                            call_expr
+                                .callee
+                                .as_member_expression()
+                                .expect("callee must be a member expression")
```

Comment:
consider returning `fixer.noop()` if callee is not a member expression (just in case)

Reply:
yeah i changed it to a debug asserd. both of these `expect`s should be unreachable

Reply:
I don't see the `debug_assert`. Either way, please make this change. Things happen that we don't expect :)

Reply:
apologies, forgot to push 😅